main/src/main/scala/sbt/Defaults.scala

```diff
@@ -2310,10 +2310,11 @@ object Defaults extends BuildCommon {
     val ci = (compile / compileInputs).value
     val ping = earlyOutputPing.value
     val reporter = (compile / bspReporter).value
+    val prevAnalysis = previousCompile.value.analysis.toOption.getOrElse(Analysis.empty)
```

Comment:
I'm concerned about the compile performance degradation by depending on `previousCompile` in `compileIncrementalTask` 🤔